etc/correctFieldMapJsonCase4.py

```diff
@@ -0,0 +1,69 @@
+#! /usr/bin/env python3
+'''
+correct field map json
+'''
+ 
+import os
+import sys
+import json
+import glob
+import collections
+
+def correctFieldMapJson(bids_dir,sub,ses=None):
+    
+    if ses: #ses not None
+        sub_prefix = '{}_{}'.format(sub,ses)
+        sub_path_prefix=os.path.join(sub,ses)
+        sub_root = '{}'.format(sub)
+    else:
+        sub_prefix = '{}'.format(sub)
+        sub_path_prefix = sub_prefix
+        sub_root = '{}'.format(sub)
+        
+    sub_dir=os.path.join(bids_dir,sub_path_prefix)
+    sub_root_dir=os.path.join(bids_dir,sub_root) #without session
+
+    for fmri_json_file in glob.glob(os.path.join(sub_dir,'fmap','{}_acq-EPI_dir-*_epi.json'.format(sub_prefix))):
+
+        #debug
+        print(fmri_json_file)
+
+        #load json files
+        with open(fmri_json_file, 'r') as f:
+            fmri_json = json.load(f,object_pairs_hook=collections.OrderedDict)
+
+        # apply to all bold images
+        cwd=os.getcwd()
```

Comment:
I guess we're changing the working directory to get the paths relative to the participant directory? I try to avoid changing directories if possible (can cause problems e.g. if the script fails or is cancelled part way through) but if this is the easiest way to do it, that's fine.